### DISCUSSION TITLE
Welcome to moodle-tiny_cloze Discussions!

### DISCUSSION THREAD

**Discussion by srobotta** [srobotta/moodle-tiny_cloze#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by srobotta** [comment]
Hi folks,

This is the first time that I am using the discussions here at github. Let's see how this works and whether this suits our needs.

After moodle moot DACH I have had some time to do more development. The dialogue boxes are working and the text labels appear. Also the events on the elements in the dialogue modal should work.
The basic idea on how the edit function of an existing question works is the following:
- onInit() parses the complete editor content and places <span> elements with a specific class around the questions.
- because of these spans the start and end of the question can be easily observed (the hard work is done in addMarkers()).
- the question text is not editable anymore in the WYSIWYG mode, can only modified via the dialogue. For those who want to edit the question text directly need to use the source code mode of the editor.

What does not yet work:
- replacement of the edited and modified question, new question text is not inserted.
- new created question is inserted at the wrong place, furthermore I think the response text is not inserted.
- feedback text has not been tested.
- delete and add icons in the dialogue to modifiy a question do not work.
- checkout whether this crypto.randomUUID() might have to be checked and encapsulated for compatibility reasons.
- behat tests (to fully test the functionality test steps need to be extended) - I think Marcus did some work already.
- maybe style the <span> a little more (e.g. with a background color) to show the user that this is not editable (once he clicks, he will realize this). This could also be configurable via the settings (tiny-multilang2 has something similar).
- check for the need of the blur stuff. I copied this over from tiny-multilang2 becaus there was the problem that when you left the editor but still remained on the page itself, the styling got lost (i.e. the encapsulating <span> tags would be gone). It feels like you are in soure code mode. To get them back switch to source code view and back to wysiwyg to have them again. I guess we have the same issue here but I haven't tested this.

I guess I will not be able to work on this during this week. All of you who were present at moode moot DACH 23 have access to the repo. Please feel free to add, change, refactor things to make the plugin work.

It was a  great time with you guys working on this. Let's keep the project running to get to a working version.

@Freakrider if you see weird things in the js, feel free to refactor this, I am also happy if you drop a note why you change things, so that we others may learn from your knowledge.

Best regards, Stephan

**Comment by marcusgreen** [comment]
Hi Stephan 
I updated my code with yours last night, nice progress. Will look again as soon as I can and see if there is anything I can contribute.

**Comment by marcusgreen** [comment]
i have just had a very quick look at the progress, I am very impressed.

**Comment by marcusgreen** [comment]
I love the way there are screen shots in the readme. Excellent work.

**Reply by srobotta** [reply]
Thank you. I need them for registering the plugin at moodle.org. So I thought I can add them here in the repo and use them for documentation purposes at the same time.
